idaes/core/util/model_diagnostics.py

```diff
@@ -3887,6 +3930,51 @@ def b_rule(b, i):
     return ill_cond
 
 
+def compute_infeasibility_explanation(
+    model, solver, tee=False, tolerance=1e-6, stream=None
+):
+    """
+    This function attempts to determine why a given model is infeasible. It deploys
+    two main algorithms:
+
+    1. Relaxes the constraints of the problem, and reports to the user
+       some sets of constraints and variable bounds, which when relaxed, creates a
+       feasible model.
+    2. Uses the information collected from (1) to attempt to compute a Minimal
+       Infeasible System (MIS), which is a set of constraints and variable bounds
+       which appear to be in conflict with each other. It is minimal in the sense
+       that removing any single constraint or variable bound would result in a
+       feasible subsystem.
+
+    This function is a wrapper for the same capability in pyomo.contrib.iis.mis.
```

Comment:
You will need to add a few exclusions to the spell-checker file as well.

Reply:
Done in [34668b0](https://github.com/IDAES/idaes-pse/pull/1409/commits/34668b0d30b1dac4c0fe905abf11e73b355a48a4)